### ISSUE TITLE
Confusing/conflicting info about how to use solarized themes

### ISSUE THREAD

**Issue by alphapapa** [Bruce-Connor/smart-mode-line#120]
I use `solarized-dark`, currently version `20150308.128`. 

The smart-mode-line readme says that the `automatic` theme is "highly recommended", but when I use that theme, it doesn't seem to use the appropriate colors.  When I use `respectful`, it does (at least, I think it does, but they don't quite match the screenshot).

`automatic`: ![automatic](http://i.imgur.com/54Kw2gw.png)
`respectful`: ![respectful](http://i.imgur.com/3UcEAHs.png)

Also, `automatic` is not available as a selection when using `sml/apply-theme`.  If it's really the recommended one, shouldn't it be a choice?

Thanks, I really like smart-mode-line.  :)

**Comment by Bruce-Connor** [comment]
> but when I use that theme

How are you using the theme exactly?
Your first image looks quite strange indeed. `automatic` is supposed to decide between setting the theme to `light`, `dark`, or `nil`, but your screenshot doesn't display any of those.

- What is the value of your `custom-enabled-themes` variable?
- What about the `sml/theme` variable?
- Does it help if you make sure `(sml/setup)` is the very last thing in your init file?

**Comment by alphapapa** [comment]
My init file has:
```
(custom-set-variables
 ; ...
 '(custom-enabled-themes (quote (solarized-dark)))
```
`sml/theme` is not in my init file.

I tried putting `(sml/setup)` at the very end of the init file, and after restarting Emacs, `describe-variable` says:
```
sml/theme is a variable defined in `smart-mode-line.el'.
Its value is dark
Original value was automatic
```
When I use `sml/apply-theme` and set it to `respectful`, it looks like the second screenshot.

**Comment by Bruce-Connor** [comment]
Ok, I think I see what I did wrong. 
I'll try to fix it soon. 

In the mean time, you can set sml/theme to nil before calling sml/setup. This will make sure sml doesn't use any of its themes.

**Comment by Bruce-Connor** [comment]
I'm trying to reproduce this, but I can't. Could this possibly be hapenning because one of the packages wasn't properly installed? (you were getting corrupted files when installing packages, right?)

The behavior I have is the following: With the init file I'm pasting below, I have `solarized-dark` enabled at startup, and the appearance matches the screenshot on the Readme. Furthermore, after initialization, the value of the `sml/theme` variable is `nil` (which is the expected here).

```
(package-initialize)
(custom-set-variables
 '(custom-enabled-themes (quote (solarized-dark)))
 '(custom-safe-themes
   (quote     ("..." ... default))))
(sml/setup)
```

If this doesn't work for you (using exactly the lines above as your init file) then you might have a corrupted package. In any case, you can always try doing `(setq sml/theme nil)`  before everything else (as I mentioned above).

**Comment by alphapapa** [comment]
Thanks for your help.  I tried your suggestion and here's what happened:

Using exactly what you provided as my init file, the solarized-dark theme failed to load.  When I ran `package-list-packages`, it was not in the list of packages.

Then I added
```
(setq package-archives '(("gnu" . "http://elpa.gnu.org/packages/")
			 ("marmalade" . "http://marmalade-repo.org/packages/")
			 ("melpa" . "http://melpa.milkbox.net/packages/")))
```
to the file, and then I could see the theme in the list, and I installed it...because it indicated it wasn't installed (even though I was already using it with my other init file).  I installed the version from MELPA, with the date-stamped version number, rather than the `1.0.0` from Marmalade.  (There are also ones prefixed with `color-theme`, but I guess those are the old-style themes.  Someone really ought to deprecate those somehow, because it's a bit confusing...)

Then I restarted Emacs and it prompted me to load the theme and do so in future sessions...and then it looked like the screenshot in the readme.

So I figured that there had been a problem with the package, and now that it was corrected, my regular init file should work.  So I switched back to that init file...but now it looks like it used to look, before I upgraded Emacs to 24.4...which is nice, yet still different than the other two screenshots, and not how it's supposed to look:

![](https://cloud.githubusercontent.com/assets/601365/6652581/5b6606c8-ca41-11e4-9940-2d02a9e1c947.png)

On top of that, now when I run `sml/setup`, nothing happens, and there is no `sml/apply-theme` command, though there are `sml/{setup,customize,bug-report,customize-faces,toggle-shorten-modes,toggle-shorten-directory}`.

I don't know what's going on.  Maybe it has something to do with different versions of different packages being installed from different repos...but I thought Emacs was supposed to load the latest version automatically, since it doesn't automatically remove obsolete versions.

I'm wondering if I should just start from scratch and use something like Cask/Pallet, though I really don't want to have to do that right now...

Thanks for any further advice.

**Comment by Bruce-Connor** [comment]
Ok, first a clarification:

The package manager does indeed propertize the most recent version of a package, but that only applies to packages installed with it. It has no control over packages you install manually. That's why you were getting an old version.

---

As for smart-mode-line, the final behaviour you describe now sounds correct. 

-When sml/setup is called, if the solarized theme is already loaded you'll get the screenshot in the Readme.
- If sml/setup is called while the solarized theme is **not** active, you'll get the dark theme (which is what you're getting).

**Comment by alphapapa** [comment]
Yep, I had totally forgotten about having installed that old version manually.  PEBKAC  :)

As for the sml problem, I just reorganized my whole init file using `outline-mode` with `outshine`, and now this problem seems to have disappeared.  When starting Emacs and running `sml/setup`, it looks like the screenshot in the readme.

I also just discovered that apparently an old `emacs-color-theme-solarized` that I had installed manually into `~/.emacs.d/themes` was being loaded by the init file.  I guess it was conflicting with the new `load-theme` style theme, because now several other things in Emacs look slightly different.

So I guess in the end this all boils down to a PEBKAC caused by my manually installing packages way-back-when and forgetting about it, and not noticing in my init file.  When I reorganized it, I deleted stuff like that, and now it's all fixed.  :)

Thanks for your help, and sorry for wasting your time on a problem that wasn't a bug at all.  :)
